components/dns_challenge/main.rs

```diff
@@ -0,0 +1,72 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#![allow(dead_code)]
+
+// Needed to derive `Serialize` on ServiceProperties
+#![feature(custom_derive, plugin)]
+#![plugin(serde_macros)]
+
+// Make linter fail for every warning
+#![plugin(clippy)]
+#![deny(clippy)]
+// Needed for many #[derive(...)] macros
+#![allow(used_underscore_binding)]
+
+#![feature(associated_consts)]
```

Comment:
can you double check what you need exactly? It looks like `#![allow(used_underscore_binding)]` and `#![feature(associated_consts)]` may not be needed